### ISSUE TITLE
Lisp error when translating assume

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1906]
Imported from SourceForge on 2024-07-05 10:27:46
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2020-07-16 12:08:46
Original: https://sourceforge.net/p/maxima/bugs/3642

---

Translating the function larry gives a Lisp error:

~~~
(%i1)	larry(e,fct) := block([ctx : supcontext(gensym(), 'initial)],
	       assume(fct),
	        e : expand(e,0,0),
	       killcontext(ctx),
	       e)$

(%i3)	translate(larry);
Maxima encountered a Lisp error:

~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:27:48
Created by **[kjak](https://sourceforge.net/u/kjak/)** on 2020-07-16 12:45:34
Original: https://sourceforge.net/p/maxima/bugs/3642/#b31a

---

- **summary**: Translating a function gives a Lisp error: --> Lisp error when translating assume
- **assigned_to**: Kris Katterjohn

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:27:52
Created by **[kjak](https://sourceforge.net/u/kjak/)** on 2020-07-16 12:45:34
Original: https://sourceforge.net/p/maxima/bugs/3642/#41e0

---

Thanks for the report.

I just took a quick glance and it looks like the translate property for `assume` makes an incorrect assumption (heh) about the form of its arguments.  I should be able to take a closer look and fix this up soon.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:27:55
Created by **[kjak](https://sourceforge.net/u/kjak/)** on 2020-07-16 18:08:22
Original: https://sourceforge.net/p/maxima/bugs/3642/#d25b

---

This has been fixed by commit [d04f9f290ec930d39076b153a8ac163a5b791d3c].
